### PR TITLE
adds envio as supported data indexer

### PR DESCRIPTION
**Pull request by SvenMuller95**
waiting for issue https://github.com/kkrt-labs/kakarot-rpc/issues/1333 to finish syncing Hypersync

### CODE REVIEW

**Comment by SvenMuller95** [comment]
This has been solved, and Hypersync has been fully onboarded onto Sepolia Testnet and is at chain head: 
https://kakarot-sepolia.hypersync.xyz/height

tl;dr 
- Hypersync is automatically used in the indexer as a more performant data source than standard RPC for data retrieval (up to 1000 times faster).  Standard RPC can still be used as well if desired (e.g. app with a small amount of data). 
- Hypersync is a low-level API with clients in python, rust, nodejs, and go for data retrieval, and also comes with a RPC interface (for projects that have existing rpc code but want to increase their performance). Great for dapp building, analytics, explorers, wallets, etc.

File: docs/ecosystem/data-indexers/envio.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+sidebar_position: 3
+---
+
+Envio is a modern, multi-chain EVM blockchain indexing framework speed-optimized for querying real-time and historical data.
+
+### Understanding Envio
+
+#### Envio HyperIndex
+
+Envio [HyperIndex](https://docs.envio.dev/docs/overview) is a feature-rich indexing solution that provides Kakarot ZK-EVM projects with a seamless and efficient way to index and aggregate real-time or historical blockchain data. The indexed data is easily accessible through custom GraphQL queries, giving developers the flexibility and power to retrieve specific information for their blockchain application.
```
Comment (Eikix) [inline docs/ecosystem/data-indexers/envio.md]:
```suggestion
Envio [HyperIndex](https://docs.envio.dev/docs/overview) is a feature-rich indexing solution that provides Kakarot Starknet projects with a seamless and efficient way to index and aggregate real-time or historical blockchain data. The indexed data is easily accessible through custom GraphQL queries, giving developers the flexibility and power to retrieve specific information for their blockchain application.
```

File: docs/ecosystem/data-indexers/envio.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Envio offers native support for Kakarot ZK-EVM networks and has been designed to support high-throughput blockchain applications that rely on real-time data for their business requirements.
```
Comment (Eikix) [inline docs/ecosystem/data-indexers/envio.md]:
```suggestion
Envio offers native support for Kakarot Starknet networks and has been designed to support high-throughput blockchain applications that rely on real-time data for their business requirements.
```

File: docs/ecosystem/data-indexers/envio.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Designed to optimize the developer experience, Envio offers automatic code generation, flexible language support, quickstart templates, and a reliable cost-effective [hosted service](https://docs.envio.dev/docs/hosted-service). Indexers on Envio can be written in JavaScript, TypeScript, or ReScript.
+
+#### Envio HyperSync
+
+Envio [HyperSync](https://docs.envio.dev/docs/overview-hypersync) is adding support on Kakarot ZK-EVM testnet sepolia and to-be mainnet.
```
Comment (Eikix) [inline docs/ecosystem/data-indexers/envio.md]:
```suggestion
Envio [HyperSync](https://docs.envio.dev/docs/overview-hypersync) is adding support on Kakarot Starknet Sepolia and to-be Kakarot Starknet mainnet. 
```

File: docs/ecosystem/data-indexers/envio.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+Envio offers native support for Kakarot ZK-EVM networks and has been designed to support high-throughput blockchain applications that rely on real-time data for their business requirements.
+
+Designed to optimize the developer experience, Envio offers automatic code generation, flexible language support, quickstart templates, and a reliable cost-effective [hosted service](https://docs.envio.dev/docs/hosted-service). Indexers on Envio can be written in JavaScript, TypeScript, or ReScript.
+
+#### Envio HyperSync
+
+Envio [HyperSync](https://docs.envio.dev/docs/overview-hypersync) is adding support on Kakarot ZK-EVM testnet sepolia and to-be mainnet. 
+
+HyperSync is a real-time data query layer for Kakarot ZK-EVM, providing APIs that bypass traditional JSON-RPC for up to 1000x faster syncing of historical data. HyperSync is used by default in Envio's indexing framework (HyperIndex), with RPC being optional for data retrieval.
```
Comment (Eikix) [inline docs/ecosystem/data-indexers/envio.md]:
```suggestion
HyperSync is a real-time data query layer for Kakarot Starknet, providing APIs that bypass traditional JSON-RPC for up to 1000x faster syncing of historical data. HyperSync is used by default in Envio's indexing framework (HyperIndex), with RPC being optional for data retrieval. 
```

File: docs/ecosystem/data-indexers/envio.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+Designed to optimize the developer experience, Envio offers automatic code generation, flexible language support, quickstart templates, and a reliable cost-effective [hosted service](https://docs.envio.dev/docs/hosted-service). Indexers on Envio can be written in JavaScript, TypeScript, or ReScript.
+
+#### Envio HyperSync
+
+Envio [HyperSync](https://docs.envio.dev/docs/overview-hypersync) is adding support on Kakarot ZK-EVM testnet sepolia and to-be mainnet. 
+
+HyperSync is a real-time data query layer for Kakarot ZK-EVM, providing APIs that bypass traditional JSON-RPC for up to 1000x faster syncing of historical data. HyperSync is used by default in Envio's indexing framework (HyperIndex), with RPC being optional for data retrieval. 
+
+Using HyperSync, Kakarot ZK-EVM projects do not need to worry about RPC URLs, rate-limiting, or managing their infrastructure - and can easily sync large datasets in a few minutes, something that would usually take hours or days using traditional indexing solutions.
```
Comment (Eikix) [inline docs/ecosystem/data-indexers/envio.md]:
```suggestion
Using HyperSync, Kakarot Starknet projects do not need to worry about RPC URLs, rate-limiting, or managing their infrastructure - and can easily sync large datasets in a few minutes, something that would usually take hours or days using traditional indexing solutions. 
```

**Review by Eikix — APPROVED** [review]

**Pull request merged**